charmcraft.yaml

```diff
@@ -0,0 +1,8 @@
+type: charm
```

Comment:
I am not 100% sure this should be committed. @jnsgruk ?

Reply:
Yeh this should definitely be committed; it will soon be mandatory in a later version of `charmcraft`. It's sparse at the moment, but will ultimately control the architectures the charm is built/run on.